src/Sentry.Unity.Editor/Native/SentryWindowsPlayer.cs

```diff
@@ -0,0 +1,140 @@
+            UseShellExecute = false,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+            CreateNoWindow = true
+        };
+
+        internal static void AddNativeOptions(SentryUnityOptions options)
+        {
+        }
+
+        internal static void AddSentryToMain(SentryUnityOptions options)
+        {
+        }
+
+        internal static string LocateWindowsPlayerSource(IEditorApplication? editorApplication = null)
+        {
+            editorApplication ??= EditorApplicationAdapter.Instance;
+
+            var playerProjectPath = Path.Combine(editorApplication.ApplicationContentsPath, "PlaybackEngines", "windowsstandalonesupport", "source", "windowsplayer");
+            if (!Directory.Exists(playerProjectPath))
+            {
+                throw new DirectoryNotFoundException($"Failed to locate the WindowsPlayer source at {playerProjectPath}.");
+            }
+
+            return playerProjectPath;
+        }
+
+        internal static void CreateWindowsPlayerProject(string windowsPlayerSource, string windowsPlayerTarget, IDiagnosticLogger? logger)
+        {
+            EditorFileIO.CopyDirectory(windowsPlayerSource, windowsPlayerTarget, logger);
+
+            // TODO: Does the .props file have to look like that?
+            using var props = File.CreateText(Path.Combine(windowsPlayerTarget, "UnityCommon.props"));
+            props.Write(@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+</Project>");
+        }
+
+        internal static string LocateMSBuild(string vsWherePath, IDiagnosticLogger? logger)
+        {
+            StartInfo.FileName = vsWherePath;
+            StartInfo.Arguments = "-latest -requires Microsoft.Component.MSBuild -find MSBuild\\**\\Bin\\MSBuild.exe";
+
+            var vsWhereOutput = "";
+            var process = new Process { StartInfo = StartInfo };
+            process.OutputDataReceived += (sender, args) => vsWhereOutput += args.Data;
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+
+            logger?.LogDebug("VSWhere returned with: {0}", vsWhereOutput);
+
+            if (!File.Exists(vsWhereOutput))
+            {
+                throw new FileNotFoundException($"Failed to locate 'msbuild'. VSWhere returned {vsWhereOutput}");
+            }
+
+            return vsWhereOutput;
+        }
+
+        internal static string LocateVSWhere(IDiagnosticLogger? logger)
+        {
+            var projectPath = Path.GetDirectoryName(Application.dataPath);
+            var directories = Directory.GetDirectories(Path.Combine(projectPath, "Library", "PackageCache"), "com.unity.ide.visualstudio@*");
+            if (directories is null || directories.Length < 1)
+            {
+                throw new Exception("Failed lo locate the 'com.unity.ide.visualstudio' package.");
```

Comment:
Some other type other than the base is preferred:

```suggestion
                throw new InvalidOperationException("Failed lo locate the 'com.unity.ide.visualstudio' package.");
```